applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_potential_flow_element.cpp

```diff
@@ -0,0 +1,839 @@
+    }
+}
+
+template <int Dim, int NumNodes>
+void IncompressiblePotentialFlowElement<Dim, NumNodes>::AssignLocalSystemWakeElement(
+    MatrixType& rLeftHandSideMatrix, Matrix& lhs_total, const ElementalData<NumNodes, Dim>& data)
+{
+    for (unsigned int row = 0; row < NumNodes; ++row)
+        AssignLocalSystemWakeNode(rLeftHandSideMatrix, lhs_total, data, row);
+}
+
+template <int Dim, int NumNodes>
+void IncompressiblePotentialFlowElement<Dim, NumNodes>::AssignLocalSystemWakeNode(
+    MatrixType& rLeftHandSideMatrix,
+    Matrix& lhs_total,
+    const ElementalData<NumNodes, Dim>& data,
+    unsigned int& row)
+{
+    // Filling the diagonal blocks (i.e. decoupling upper and lower dofs)
+    for (unsigned int column = 0; column < NumNodes; ++column)
+    {
+        rLeftHandSideMatrix(row, column) = lhs_total(row, column);
+        rLeftHandSideMatrix(row + NumNodes, column + NumNodes) = lhs_total(row, column);
+    }
+
+    // Applying wake condition on the AUXILIARY_VELOCITY_POTENTIAL dofs
+    if (data.distances[row] < 0.0)
+        for (unsigned int column = 0; column < NumNodes; ++column)
+            rLeftHandSideMatrix(row, column + NumNodes) = -lhs_total(row, column); // Side 1
+    else if (data.distances[row] > 0.0)
+        for (unsigned int column = 0; column < NumNodes; ++column)
+            rLeftHandSideMatrix(row + NumNodes, column) = -lhs_total(row, column); // Side 2
+}
+
+template <int Dim, int NumNodes>
+void IncompressiblePotentialFlowElement<Dim, NumNodes>::CheckWakeCondition()
+{
+    array_1d<double, Dim> upper_wake_velocity;
+    ComputeVelocityUpperWakeElement(upper_wake_velocity);
+    const double vupnorm = inner_prod(upper_wake_velocity, upper_wake_velocity);
+
+    array_1d<double, Dim> lower_wake_velocity;
+    ComputeVelocityLowerWakeElement(lower_wake_velocity);
+    const double vlownorm = inner_prod(lower_wake_velocity, lower_wake_velocity);
+
+    if (std::abs(vupnorm - vlownorm) > 0.1)
+        std::cout << "WAKE CONDITION NOT FULFILLED IN ELEMENT # " << this->Id()
+                  << std::endl;
+}
+
+template <int Dim, int NumNodes>
+void IncompressiblePotentialFlowElement<Dim, NumNodes>::ComputePotentialJump(ProcessInfo& rCurrentProcessInfo)
+{
+    const array_1d<double, 3> vinfinity = rCurrentProcessInfo[VELOCITY_INFINITY];
+    const double vinfinity_norm = sqrt(inner_prod(vinfinity, vinfinity));
+
+    array_1d<double, NumNodes> distances;
+    GetWakeDistances(distances);
+
+    for (unsigned int i = 0; i < NumNodes; i++)
+    {
+        double aux_potential = GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        double potential = GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+        double potential_jump = aux_potential - potential;
+
+        if (distances[i] > 0)
+        {
+            GetGeometry()[i].SetValue(POTENTIAL_JUMP, 2.0 / vinfinity_norm * (potential_jump));
```

Comment:
Hi @inigolcanalejo ,

Is there a missing "-" here?
```suggestion
            GetGeometry()[i].SetValue(POTENTIAL_JUMP, -2.0 / vinfinity_norm * (potential_jump));
```

Reply:
Hi @marcnunezc, thanks for your comment, the short answer is I don't think so. Let me give a few comments about this line:

1. Actually the variable POTENTIAL_JUMP here is not really the potential_jump but the lift coefficient.
2. The lift coefficient can be computed using the following implemented formula:
![formula_cl](https://user-images.githubusercontent.com/28628414/51832462-0c49b600-22f5-11e9-87b9-24d2308f5b1a.png)

Having said this, why do you think there is a missing "-"?

Reply:
Hi!

I have tried to compute the lift coefficient in the same way you show here in my branch and I have seen that if a node has a **positive distance** then:

 `VELOCITY_POTENTIAL>AUXILIARY_VELOCITY_POTENTIAL`

and as you defined 
`potential_jump= aux_potential-potential`
the potential jump (and the Cl as you pointed out) would give a negative value (that's why i suggested adding a "-"). Another option could be defining the potential_jump as "potential-aux_potential" or "aux_potential-potential" depending on the distance value.

Do you agree with this?

Reply:
You are right, actually there is a minus missing there. In the original implementation I had it as you suggest, but then I changed it to reduce the length of the line and I forgot to put the minus. Ill commit your suggestion ;-) Thanks for the review!